nidata/multimodal/hcp/datasets.py

```diff
@@ -14,6 +14,8 @@ def fetch(self, files, force=False, resume=True, check=False, verbose=1):
         if self.jsession_id is None:
             # Log in to the website.
             import requests
+            self.username = os.environ.get("NIDATA_USERNAME")
+            self.passwd = os.environ.get("NIDATA_PASSWD")
```

Comment:
`self.username` is set inside `__init__`, so I think this should happen there. Also, if `username` was specified in `__init__`, it gets overwritten here. I'd rather that these environment variables are only used *if `username` and/or `password` are not passed directly in `__init__.